File: lib/utilities/serviceIpRange.ts
import * as ec2 from "aws-cdk-lib/aws-ec2";

import * as fs from "fs";
import * as path from "path";

interface IpRanges {
  syncToken: string;
  createDate: string;
  prefixes: {
    ip_prefix: string;
    region: string;
    service: string;
    network_border_group: string;
  }[];
  ipv6_prefixes: {
    ipv6_prefix: string;
    region: string;
    service: string;
    network_border_group: string;
  }[];
}

const data = JSON.parse(
  fs.readFileSync(path.resolve(__dirname, "./ip-ranges.json")).toString()
) as IpRanges;

export default function serviceIpRange(service: string, region: string) {
  return data.prefixes
    .filter((entry) => entry.service === service)
    .filter((entry) => entry.region === region)
    .map((entry) => entry.ip_prefix)[0];
}

File: nitecap/utilities/service_ip_range.py
import json
import requests


def service_ip_range(service: str, region: str):
    aws_ip_ranges = json.loads(
        requests.get("https://ip-ranges.amazonaws.com/ip-ranges.json").text
    )

    for entry in aws_ip_ranges["prefixes"]:
        if entry["service"] == service and entry["region"] == region:
            return entry["ip_prefix"]

    raise NameError
